### DISCUSSION TITLE
How to hide a form field with dehydrated?

### DISCUSSION THREAD

**Discussion by NaungYeHtet** [filamentphp/filament#11279] [Help] [answered]
### Package

Form builder

### Package Version

v3.0.0

### How can we help you?

```
Forms\Components\TextInput::make('type')
    ->default(MyType::from('some')->value)
    ->hidden()
    ->dehydrated(),
```
I want to hide the field and also want to include in form data when create. How can I achieve this?

**Comment by leandrocfe** [comment]
Why not [customizing data before saving](https://filamentphp.com/docs/3.x/panels/resources/creating-records#customizing-data-before-saving) ?

**Reply by StaffCollab** [reply, marked as answer]
If `mutateFormDataBeforeCreate()` or these methods are not working for you. (it didn't work for me since I was in an Attach action, and I also have some other unique use cases) There is a simple trick to achieve the same results

Just use a `Hidden::make('name')->default('John')->dehydrated(true)` field. This will work, and if you optionally do need to customize the data, you can create a regular `TextInput::make()` field, and it will be synced with the hidden one automatically since both use the same `wire:model` behind the scene

**Comment by mrksdiehl** [comment]
You could also use CSS to hide the field. With tailwind I did this `->extraFieldWrapperAttributes(['class' => 'h-0 opacity-0'])`. That's how I solved it with an external component where I wanted to keep the logic it implemented. Worked for me.
